bom/application/pom.xml

```diff
@@ -98,7 +98,9 @@
         <jboss-transaction-api_1.2_spec.version>1.1.1.Final</jboss-transaction-api_1.2_spec.version>
         <agroal.version>1.14</agroal.version>
         <jboss-transaction-spi.version>7.6.0.Final</jboss-transaction-spi.version>
-        <elasticsearch-rest-client.version>7.10.0</elasticsearch-rest-client.version>
+        <elasticsearch-opensource-components.version>7.16.2</elasticsearch-opensource-components.version>
+        <!-- for the now proprietary components of Elasticsearch, we use the last Open Source version -->
+        <elasticsearch-proprietary-components-keeping-old-opensource-version.version>7.10.0</elasticsearch-proprietary-components-keeping-old-opensource-version.version>
```

Comment:
I believe users usually change the version of a dependency by overriding a property; if you change the name of the property, it will affect existing applications in unexpected ways.

I guess we could do something like this, though it will only solve the problem for *some* users. Maybe that's why you didn't even try?

```suggestion
        <elasticsearch-rest-client.version>7.16.2</elasticsearch-rest-client.version>
        <elasticsearch-opensource-components.version>${elasticsearch-rest-client.version}</elasticsearch-opensource-components.version>
        <!-- for the now proprietary components of Elasticsearch, we use the last Open Source version -->
        <elasticsearch-proprietary-components-keeping-old-opensource-version.version>7.10.0</elasticsearch-proprietary-components-keeping-old-opensource-version.version>
```